dbms/src/Core/Block.cpp

```diff
@@ -514,6 +514,26 @@ static ReturnType checkBlockStructure(const Block & lhs, const Block & rhs, cons
     return ReturnType(true);
 }
 
+Block blocksMerge(Blocks && result_blocks)
```

Comment:
Better to reuse the result_blocks memory instead of cloneEmpty every time. Maybe can take SquashingBlockInputStream as an example.

Reply:
> Better to reuse the result_blocks memory instead of cloneEmpty every time. Maybe can take SquashingBlockInputStream as an example.

done.